File: mobile/lib/models/product.dart
class Product {
  final String id;
  final String name;
  final String category;
  final double price;
  final String? image;
  final String? description;
  final int stock;

  Product({
    required this.id,
    required this.name,
    required this.category,
    required this.price,
    this.image,
    this.description,
    required this.stock,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      category: json['category'] ?? 'General',
      price: (json['price'] ?? 0).toDouble(),
      image: json['image'],
      description: json['description'],
      stock: json['stock'] ?? 0,
    );
  }
}

File: backend/model/Product.js
// Product model for MongoDB
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
  name: { type: String, required: true },
  category: { type: String, default: 'General' },
  price: { type: Number, required: true },
  image: { type: String },
  description: { type: String },
  stock: { type: Number, default: 0 }
});

module.exports = mongoose.model('Product', productSchema);
